xskillscore/core/probabilistic.py

```diff
@@ -550,48 +568,120 @@ def rps(
     * C. A. T. Ferro. Fair scores for ensemble forecasts. Q.R.J. Meteorol. Soc., 140:
       1917–1923, 2013. doi: 10.1002/qj.2270.
     * https://www-miklip.dkrz.de/about/problems/
+
     """
     bin_names = ["category"]
-    M = forecasts[member_dim].size
     bin_dim = f"{bin_names[0]}_bin"
-    # histogram(dim=[]) not allowed therefore add fake member dim
-    # to apply over when multi-dim observations
-    if len(observations.dims) == 1:
-        observations = histogram(
-            observations, bins=[category_edges], bin_names=bin_names, dim=None
-        )
-    else:
-        observations = histogram(
-            observations.expand_dims(member_dim),
+    M = forecasts[member_dim].size
+
+    def _bool_to_int(ds):
+        """convert xr.object of dtype bool to int to evade:
+        TypeError: numpy boolean subtract, the `-` operator, is not supported"""
+
+        def _helper_bool_to_int(da):
+            if da.dtype == "bool":
+                da = da.astype("int")
+            return da
+
+        if isinstance(ds, xr.Dataset):
+            ds = ds.map(_helper_bool_to_int)
+        else:
+            ds = _helper_bool_to_int(ds)
+        return ds
+
+    forecasts = _bool_to_int(forecasts)
+
+    def _check_identical_xr_types(a, b):
```

Comment:
I imagine this could go higher in our code base: https://github.com/xarray-contrib/xskillscore/issues/29

Reply:
I'd also move this to utils. can be applied to basically all metrics?

Reply:
can be done in PR closing #29